policy-controller/k8s/index/src/outbound_index.rs

```diff
@@ -0,0 +1,361 @@
+                        tracing::warn!(?parent_ref, "ignoring parent_ref with port 0");
+                    }
+                } else {
+                    tracing::warn!(?parent_ref, "ignoring parent_ref without port");
+                }
+            } else {
+                tracing::warn!(
+                    parent_kind = parent_ref.kind.as_deref(),
+                    "ignoring parent_ref with non-Service kind"
+                );
+            }
+        }
+    }
+
+    fn update_service(&mut self, name: String, service: ServiceInfo) {
+        tracing::debug!(?name, ?service, "updating service");
+        for (svc_port, svc_routes) in self.service_routes.iter_mut() {
+            let opaque = service.opaque_ports.contains(&svc_port.port);
+            svc_routes.set_opaque(opaque);
+        }
+        self.services.insert(name, service);
+    }
+
+    fn delete(&mut self, name: String) {
+        for service in self.service_routes.values_mut() {
+            service.delete(&name);
+        }
+    }
+
+    fn service_routes_or_default(&mut self, service_port: ServicePort) -> &mut ServiceRoutes {
+        let authority = format!(
+            "{}.{}.svc.{}:{}",
+            service_port.service, self.namespace, self.cluster_domain, service_port.port
+        );
+        self.service_routes
+            .entry(service_port.clone())
+            .or_insert_with(|| {
+                let opaque = match self.services.get(&service_port.service) {
+                    Some(svc) => svc.opaque_ports.contains(&service_port.port),
+                    None => false,
+                };
+                let (sender, _) = watch::channel(OutboundPolicy {
+                    http_routes: Default::default(),
+                    authority,
+                    namespace: self.namespace.to_string(),
+                    opaque,
+                });
+                ServiceRoutes {
+                    routes: Default::default(),
+                    watch: sender,
+                    opaque,
+                }
+            })
+    }
+
+    fn convert_route(&self, route: HttpRoute) -> Result<OutboundHttpRoute> {
+        let hostnames = route
+            .spec
+            .hostnames
+            .into_iter()
+            .flatten()
+            .map(convert::host_match)
+            .collect();
+
+        let rules = route
+            .spec
+            .rules
+            .into_iter()
+            .flatten()
+            .map(|r| self.convert_rule(r))
+            .collect::<Result<_>>()?;
+
+        let creation_timestamp = route.metadata.creation_timestamp.map(|Time(t)| t);
+
+        Ok(OutboundHttpRoute {
+            hostnames,
+            rules,
+            creation_timestamp,
+        })
+    }
+
+    fn convert_rule(&self, rule: HttpRouteRule) -> Result<OutboundHttpRouteRule> {
+        let matches = rule
+            .matches
+            .into_iter()
+            .flatten()
+            .map(InboundRouteBinding::try_match)
```

Comment:
Why does the outbound index use the InboundRouteBinding type? This works against modularity. shoudl try try_match be decoupled from the inbound type or?